### ISSUE TITLE
Network aliases not set

### ISSUE THREAD

**Issue by psbrandt** [docker/compose-cli#1936]
**Description**

Network aliases specified in the compose file are not correctly set.

**Steps to reproduce the issue:**

Compose file:

```yaml
version: "3"

services:
  one:
    image: hashicorp/http-echo
    ports:
      - 5678:5678
    command: -text="one"
    networks:
      default:
        aliases:
          - hello
          - hello.one
          - hello.one.local.com

  two:
    image: hashicorp/http-echo
    ports:
      - 6789:5678
    command: -text="two"
    networks:
      default:
        aliases:
          - world
          - world.two
          - world.two.local.com
```

First run `docker compose up`.

Then run `docker inspect docker-compose-bug_one_1 | jq '.[0].NetworkSettings.Networks'`

**Describe the results you received:**

```json
{
  "docker-compose-bug_default": {
    "IPAMConfig": null,
    "Links": null,
    "Aliases": [
      "one",
      "docker-compose-bug_one_1",
      "50843bd1794b"
    ],
    "NetworkID": "615b7b1ce493551ec73b79caaedeba1a3c8941d537c4b0f7b80435b7997aab73",
    "EndpointID": "c9cb31b5ac4506d99b9da922bce5a6d92da1de826c2e3aa65ff7c1372fb30904",
    "Gateway": "172.29.0.1",
    "IPAddress": "172.29.0.3",
    "IPPrefixLen": 16,
    "IPv6Gateway": "",
    "GlobalIPv6Address": "",
    "GlobalIPv6PrefixLen": 0,
    "MacAddress": "02:42:ac:1d:00:03",
    "DriverOpts": null
  }
}
```

**Describe the results you expected:**

```json
{
  "docker-compose-bug_default": {
    "IPAMConfig": null,
    "Links": null,
    "Aliases": [
      "hello.one",  
      "one",
      "hello",
      "f91c7320b6b5",
      "hello.one.local.com"
    ],
    "NetworkID": "615b7b1ce493551ec73b79caaedeba1a3c8941d537c4b0f7b80435b7997aab73",
    "EndpointID": "2bd9fc5e0103f065eec050ab0d8ede1f3d4b4bb9f58fe9748fccb5697beaad28",
    "Gateway": "172.29.0.1",
    "IPAddress": "172.29.0.3",
    "IPPrefixLen": 16,
    "IPv6Gateway": "",
    "GlobalIPv6Address": "",
    "GlobalIPv6PrefixLen": 0,
    "MacAddress": "02:42:ac:1d:00:03",
    "DriverOpts": null
  }
}
```

Note the additional aliases.

**Additional information you deem important (e.g. issue happens only occasionally):**

Setting the network aliases seems to work correctly on version `1.29.1, build c34c88b2`:

```
$ docker-compose version
docker-compose version 1.29.1, build c34c88b2
docker-py version: 5.0.0
CPython version: 3.9.0
OpenSSL version: OpenSSL 1.1.1h  22 Sep 2020
```

If I run `docker-compose up` (using version `1.29.1`), then `docker inspect docker-compose-bug_one_1 | jq '.[0].NetworkSettings.Networks'`, I get the expected result:

```json
{
  "docker-compose-bug_default": {
    "IPAMConfig": null,
    "Links": null,
    "Aliases": [
      "hello.one",  
      "one",
      "hello",
      "f91c7320b6b5",
      "hello.one.local.com"
    ],
    "NetworkID": "615b7b1ce493551ec73b79caaedeba1a3c8941d537c4b0f7b80435b7997aab73",
    "EndpointID": "2bd9fc5e0103f065eec050ab0d8ede1f3d4b4bb9f58fe9748fccb5697beaad28",
    "Gateway": "172.29.0.1",
    "IPAddress": "172.29.0.3",
    "IPPrefixLen": 16,
    "IPv6Gateway": "",
    "GlobalIPv6Address": "",
    "GlobalIPv6PrefixLen": 0,
    "MacAddress": "02:42:ac:1d:00:03",
    "DriverOpts": null
  }
}
```

**Output of `docker compose version`:**
```
Docker Compose version 2.0.0-beta.1
```

**Output of `docker version`:**

```
Client:
 Cloud integration: 1.0.14
 Version:           20.10.6
 API version:       1.41
 Go version:        go1.16.3
 Git commit:        370c289
 Built:             Fri Apr  9 22:46:57 2021
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.6
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8728dd2
  Built:            Fri Apr  9 22:44:56 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker context show`:**  
You can also run `docker context inspect context-name` to give us more details but don't forget to remove sensitive content.

```
default
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
  compose: Docker Compose (Docker Inc., 2.0.0-beta.1)
  scan: Docker Scan (Docker Inc., v0.8.0)

Server:
 Containers: 23
  Running: 4
  Paused: 0
  Stopped: 19
 Images: 202
 Server Version: 20.10.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc version: 12644e614e25b05da6fd08a38ffa0cfe1903fdec
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.25-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 15.64GiB
 Name: docker-desktop
 ID: JHVR:6J46:YTCZ:FB5C:UT5P:QLVQ:XZ7G:MDAZ:3KQQ:2TVI:ZASA:HA3X
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 79
  Goroutines: 86
  System Time: 2021-07-20T22:00:57.634157538Z
  EventsListeners: 5
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details (AWS ECS, Azure ACI, local, etc.):**

Docker for Mac

**Comment by thaJeztah** [comment]
Looks like a duplicate of https://github.com/docker/compose-cli/issues/1741, https://github.com/docker/compose-cli/issues/1620 (and some other tickets), which was fixed by https://github.com/docker/compose-cli/pull/1625

I see you're running an older version of Docker / Docker Desktop; are you still seeing this issue when updating to the current version?

**Comment by psbrandt** [comment]
I thought I upgraded via the app, but let me do a fresh download and hard re-install.

**Comment by psbrandt** [comment]
That seems to have resolved it. I can confirm the issue is fixed in Docker Compose version <kbd>v2.0.0-beta.6</kbd>.
